### PR TITLE
Validation of input parameters

### PR DESCRIPTION
**Pull request by drehak**

### CODE REVIEW

File: leapp/workflows/__init__.py [hunk 1]
```diff
@@ -178,6 +184,11 @@ def run(self, context=None, until_phase=None, until_actor=None, skip_phases_unti
 
         self._errors = get_errors(context)
 
+        for phase in skip_phases_until, needle_phase:
+            if not is_valid_phase(phase):
+                self.log.error('Phase {phase} does not exist in the workflow'.format(phase=phase))
+                return
```
Comment (vinzenz) [inline leapp/workflows/__init__.py]: well you should then ensure that we exit with non 0 status code.

File: leapp/workflows/__init__.py [hunk 2]
```diff
@@ -27,6 +27,14 @@ def _get_phases_sorted(attrs):
     return tuple(sorted([attr for attr in attrs.values() if _is_phase(attr)], key=_phase_sorter_key))
 
 
+def actor_names(actor=None):
+    return (actor.name.lower(), actor.class_name.lower()) if actor else (None)
+
+
+def phase_names(phase=None):
+    return (phase[0].__name__.lower(), phase[0].name.lower()) if phase else (None)
```
Comment (vinzenz) [inline leapp/workflows/__init__.py]:
```suggestion
    return (phase[0].__name__.lower(), phase[0].name.lower()) if phase else ()
```

File: leapp/workflows/__init__.py [hunk 3]
```diff
@@ -27,6 +27,14 @@ def _get_phases_sorted(attrs):
     return tuple(sorted([attr for attr in attrs.values() if _is_phase(attr)], key=_phase_sorter_key))
 
 
+def actor_names(actor=None):
+    return (actor.name.lower(), actor.class_name.lower()) if actor else (None)
```
Comment (vinzenz) [inline leapp/workflows/__init__.py]:
```suggestion
    return (actor.name.lower(), actor.class_name.lower()) if actor else ()
```

File: leapp/workflows/__init__.py [hunk 4]
```diff
@@ -141,6 +149,12 @@ def serialize(cls):
             'phases': [phase.serialize() for phase in cls.phases],
         }
 
+
+    def is_valid_phase(self, phase=None):
+        if phase:
+            return phase in [phase_names(phase) for phase in self._phase_actors]
```
Comment (vinzenz) [inline leapp/workflows/__init__.py]: you Need to flatten the list of tuples to be able to search in it

Comment (drehak) [reply leapp/workflows/__init__.py]: Should be fine now.

**Comment by drehak** [comment]
Since the related issue is about general validation - rather than just workflow phase names - are there any other input parameters that might need similar validation?

**Comment by vinzenz** [comment]
I am not sure if --whitelist-experimental is already validated - Please check, if not then we gotta add it too. But again It's actor names

File: leapp/workflows/__init__.py (hunk 1 above)
Comment (vinzenz) [reply leapp/workflows/__init__.py]: AFAIR yes

Comment (drehak) [reply leapp/workflows/__init__.py]: Okay then, replaced this with a CommandError.

**Comment by drehak** [comment]
Turns out it's validated. I've nuked my latest commits, hopefully nothing breaks this way.

File: leapp/workflows/__init__.py [hunk 5]
```diff
@@ -178,11 +191,15 @@ def run(self, context=None, until_phase=None, until_actor=None, skip_phases_unti
 
         self._errors = get_errors(context)
 
+        for phase in skip_phases_until, needle_phase:
+            if phase and not self.is_valid_phase(phase):
```
Comment (vinzenz) [inline leapp/workflows/__init__.py]: Is valid phase returns None if the phase is None or empty so the is unnecessary

Comment (vinzenz) [reply leapp/workflows/__init__.py]: Never mind it is necessary I just realized

**Pull request merged**